apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1063,7 +1086,14 @@ fn preprocess_errors(
             t.error.iter_mut().for_each(|err| {
                 err.message = String::from("<redacted>");
                 err.location = Vec::new();
-                err.json = String::new();
+                err.json = if matches!(
+                    error_config.redaction_policy,
+                    ErrorRedactionPolicy::Extended
+                ) {
+                    perform_exteneded_redaction(&err.json)
```

Comment:
Sorry maybe I don't understand something, but why do we keep the whole error in json instead of just the extension code as we only use extension code ? It would prevent us from deserializing it from json everytime

Reply:
I think mainly the reason is that the data model for errors in this spot comes from protobuf, and the only place to add the `code` is in the `json` field.  I don't think it's an easy refactor to change this but maybe I'm wrong.
```
    pub struct Error {
        /// required
        #[prost(string, tag = "1")]
        pub message: ::prost::alloc::string::String,
        #[prost(message, repeated, tag = "2")]
        pub location: ::prost::alloc::vec::Vec<Location>,
        #[prost(uint64, tag = "3")]
        pub time_ns: u64,
        #[prost(string, tag = "4")]
        pub json: ::prost::alloc::string::String,
    }
```

Reply:
Ah ok sorry I didn't see it was coming from protobuf.